src/main/kotlin/com/coder/gateway/CoderRemoteConnectionHandle.kt

```diff
@@ -200,5 +238,66 @@ class CoderRemoteConnectionHandle {
             }
             return Pair(tokenFromUser, tokenSource)
         }
+
+        /**
+         * Return if the URL is whitelisted, https, and the URL and its final
+         * destination, if it is a different host.
+         */
+        @JvmStatic
+        fun isWhitelisted(url: URL, deploymentURL: URL): Triple<Boolean, Boolean, String> {
+            // TODO: Setting for the whitelist, and remember previously allowed
+            //  domains.
+            val domainWhitelist = listOf("intellij.net", "jetbrains.com", deploymentURL.host)
+
+            // Resolve any redirects.
+            val finalUrl = try {
+                resolveRedirects(url)
+            } catch (e: Exception) {
+                when (e) {
+                    is SSLHandshakeException ->
+                    throw Exception(CoderGatewayBundle.message(
+                        "gateway.connector.view.workspaces.connect.ssl-error",
+                        url.host,
+                        e.message ?: CoderGatewayBundle.message("gateway.connector.view.workspaces.connect.no-reason")
+                    ))
+                    else -> throw e
```

Comment:
This may cause some headaches if the required certs are not available in the JRE keystore. I think it should be fine if it's trusted by the system though.

Reply:
From my testing (although I might have missed something), adding to the system trust store was not enough, it had to be in the JRE key store that was bundled with Gateway.

The help text has a link to the docs that explains how to add the cert, but I do think it would be nice as a future enhancement to give the option to view and accept the cert.  Not sure how difficult that would be.